### ISSUE TITLE
How to change the dataset

### ISSUE THREAD

**Issue by XinYuanZhan** [Thinklab-SJTU/ThinkMatch#47]
Hello, I have read all the papers you have published and your work is admirable! But, when I ran the code, I encountered some problems. I saw that only a few datasets are supported in the code you posted. Can I change my own dataset?

**Comment by 1115712590** [comment]
Hi, our dataset API is provided by [pygmtools](https://github.com/Thinklab-SJTU/pygmtools), which has been packed to PyPI. However, pygmtools still retains the interface to add your own datasets. You need to implement your own dataset class in [dataset.py](https://github.com/Thinklab-SJTU/pygmtools/blob/main/pygmtools/dataset.py) file.

In your dataset class, your dataset should be pre-processed to **generate 3 json files under your dataset path**.

`train.json`: list of training set, like `[object ID1, object ID2, ...]`
`test.json`: list of test set, like `[object ID1, object ID2, ...]`
`data.json`: your pre-processed dataset, and should be like:
```
{
    object ID: {
        'path': path/to/this/image
        'kpts': [{"labels":keypoint name, "x": x-coordinate after image resize, "y": y-coordinate after image resize}, ...]
	'bounds': boundbox of object
        'cls': class of object
        'univ_size': number of keypoints
    },
    ...
}
```
You can also refer to our implementation of the dataset class.

Finally, you need to write configuration files for your dataset to run the experiment as [here](https://github.com/Thinklab-SJTU/ThinkMatch#run-the-experiment).

**Comment by XinYuanZhan** [comment]
懂啦！感谢好兄弟！

**Comment by 1115712590** [comment]
You're welcome. Feel free to open new issues or reopen this issue if you have any further questions.

**Comment by XinYuanZhan** [comment]
您好！我已经按照您的方法修改了我的文件，但是始终无法生成对应的 .json 文件，我运行 train 文件时，报错显示找不到数据集的匹配项，我可能不太理解您说的 your dataset should be pre-processed to generate 3 json files under your dataset path.这个是在process函数里面吗？

**Comment by 1115712590** [comment]
Yes. In our implementation, the `process()` function is used to generate those 3 json files. You need to implement your own function to generate those json files.

**Comment by XinYuanZhan** [comment]
实在抱歉，我才刚刚开始学习深度学习。所以我只需要仿照您的 class 重新编写一个关于自己数据集的类，并且在 dataset_config.py 文件中增加相关的地址就可以了吗？我怎么单独运行 dataset 文件，来只让他生成 .json 文件呢？

**Comment by 1115712590** [comment]
The instance of the dataset class will be automatically created by `Benchmark` class, you can refer to [here](https://github.com/Thinklab-SJTU/pygmtools/blob/main/pygmtools/benchmark.py#L36). So you just need to call the function that generates the json files in your `dataset.__init__()`. You can simply create a `Benchmark` instance for your dataset for debugging.  Feel free to modify `dataset_config.py` if you think it is necessary.
